basex/src/main/java/com/formulasearchengine/mathosphere/basex/Server.java

```diff
@@ -63,19 +63,28 @@ public void startup(@NotNull File input) throws IOException {
 		shutdown();
 		baseXServer = new BaseXServer( "-p" + PORT, "-n" + SERVER_NAME ); // "-d" for debug
 		file = input;
-		final Charset charset = Charsets.UTF_8;
-		final StringBuilder stringBuilder = new StringBuilder();
-		try (BufferedReader fileReader = new BufferedReader(new InputStreamReader(new FileInputStream(file), charset))) {
-			String line = fileReader.readLine();
-			while (line != null) {
-				stringBuilder.append(line);
-				stringBuilder.append(System.getProperty("line.separator"));
-				line = fileReader.readLine();
+//		final Charset charset = Charsets.UTF_8;
+//		final StringBuilder stringBuilder = new StringBuilder();
+//		try (BufferedReader fileReader = new BufferedReader(new InputStreamReader(new FileInputStream(file), charset))) {
+//			String line = fileReader.readLine();
+//			while (line != null) {
+//				stringBuilder.append(line);
+//				stringBuilder.append(System.getProperty("line.separator"));
+//				line = fileReader.readLine();
+//			}
+//		}
+		if(file.isFile()) {
+			new CreateDB(DATABASE_NAME, file.getAbsolutePath()).execute(baseXServer.context);
+		} else {
+			File[] filesInDataDirectory = file.listFiles();
+			if(filesInDataDirectory != null) {
+				System.out.println("Importing files from directory:");
+				for(File f : filesInDataDirectory) {
+					System.out.println("Importing " + f.getAbsolutePath());
+					new CreateDB(DATABASE_NAME, f.getAbsolutePath()).execute(baseXServer.context);
```

Comment:
the documentation states
`input input reference (local/remote file or XML string; can be {@code null})`